staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_test.go

```diff
@@ -341,9 +341,18 @@ func TestSendInitialEventsBackwardCompatibility(t *testing.T) {
 }
 
 func TestCacherWatchSemantics(t *testing.T) {
-	store, terminate := testSetupWithEtcdAndCreateWrapper(t)
-	t.Cleanup(terminate)
-	storagetesting.RunWatchSemantics(context.TODO(), t, store)
+	t.Run("WatchFromStorageWithoutResourceVersion=true", func(t *testing.T) {
+		defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.WatchFromStorageWithoutResourceVersion, true)()
+		store, terminate := testSetupWithEtcdAndCreateWrapper(t)
+		t.Cleanup(terminate)
+		storagetesting.RunWatchSemantics(context.TODO(), t, store)
+	})
+	t.Run("WatchFromStorageWithoutResourceVersion=false", func(t *testing.T) {
+		defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.WatchFromStorageWithoutResourceVersion, false)()
+		store, terminate := testSetupWithEtcdAndCreateWrapper(t)
+		t.Cleanup(terminate)
+		storagetesting.RunWatchSemantics(context.TODO(), t, store)
```

Comment:
doesn't this test take a long time? does running it twice now exceed package testing time?

Reply:
Yes, it takes 42s :(

Reply:
need to fix the test in master first ... the doubling of RunWatchSemantics has made this package timeout ~50% of runs since https://github.com/kubernetes/kubernetes/pull/123935/files#r1530224210 merged

Reply:
I expect that this is related to additional tests present on the main branch. Where I can check the current test runtime?

Reply:
Sent https://github.com/kubernetes/kubernetes/pull/123994 and included it into this PR.

Reply:
Can you keep each commit picked from master distinct, and update the description with all the master PRs now included in this PR?

Reply:
There are 2 separate commits in the PR, one for https://github.com/kubernetes/kubernetes/pull/123935 and second for https://github.com/kubernetes/kubernetes/pull/123994. Updated the PR title and description to mention both PRs. Is this ok?

Reply:
> Added https://github.com/kubernetes/kubernetes/commit/92bdc7b3873800e6130176e49acdf5e17110e5b9 and https://github.com/kubernetes/kubernetes/pull/123887 to reduce test runtime

I thought you meant you were pulling in these commits to this PR... was that not the case?

Reply:
I assumed that just undoing the second run of TestWatchSemantics might get us below timeout. Pulled all the fixes to be safe.